plugins/node/opentelemetry-instrumentation-winston/src/instrumentation.ts

```diff
@@ -163,44 +186,96 @@ export class WinstonInstrumentation extends InstrumentationBase {
       ) {
-        const span = trace.getSpan(context.active());
-
-        if (!span) {
-          return original.apply(this, args);
-        }
-
-        const spanContext = span.spanContext();
+        let record: Record<string, any> = {};
+        const config = instrumentation.getConfig();
+        if (!config.disableLogCorrelation) {
+          const span = trace.getSpan(context.active());
+          if (span) {
+            const spanContext = span.spanContext();
+            if (isSpanContextValid(spanContext)) {
+              record = injectRecord(spanContext, record);
+            }
+            instrumentation._callHook(span, record);
 
-        if (!isSpanContextValid(spanContext)) {
-          return original.apply(this, args);
-        }
+            // Inject in metadata argument
+            let isDataInjected = false;
+            for (let i = args.length - 1; i >= 0; i--) {
+              if (typeof args[i] === 'object') {
+                args[i] = Object.assign(args[i], record);
+                isDataInjected = true;
+                break;
+              }
+            }
+            if (!isDataInjected) {
+              const insertAt =
+                typeof args[args.length - 1] === 'function'
+                  ? args.length - 1
+                  : args.length;
 
-        for (let i = args.length - 1; i >= 0; i--) {
-          if (typeof args[i] === 'object') {
-            const record = args[i];
-            injectRecord(spanContext, record);
-            instrumentation._callHook(span, record);
-            return original.apply(this, args);
+              args.splice(insertAt, 0, record);
+            }
           }
         }
 
-        const record = injectRecord(spanContext);
+        if (!config.disableLogSending) {
+          const logRecord: Record<string, any> = {};
+          if (args.length >= 1) {
+            // Level is always first argument
+            logRecord['level'] = args[0];
 
-        const insertAt =
-          typeof args[args.length - 1] === 'function'
-            ? args.length - 1
-            : args.length;
+            // Get meta if present
+            for (let i = args.length - 1; i >= 0; i--) {
+              if (typeof args[i] === 'object') {
+                logRecord['meta'] = args[i];
+                break;
+              }
+            }
 
-        args.splice(insertAt, 0, record);
-        instrumentation._callHook(span, record);
+            let msgArguments = 0;
+            for (let i = 1; i < args.length; i++) {
+              if (typeof args[i] === 'string') {
+                msgArguments++;
+              }
+            }
+            if (msgArguments > 0) {
+              if (msgArguments === 1) {
+                logRecord['msg'] = args[1];
+              } else {
+                // Handle string interpolation
+                const values = args.slice(2, msgArguments + 1);
+                logRecord['msg'] = util.format(args[1], ...values);
```

Comment:
I didn't run it myself, but it seems [winston describes that this behavior is not always applied](https://github.com/winstonjs/winston#string-interpolation), and if it does, the arguments might be non-string:

> The log method provides the string interpolation using [util.format](https://nodejs.org/dist/latest/docs/api/util.html#util_util_format_format_args). It must be enabled using format.splat().

and: 

```js
logger.log('info', 'test message %d', 123);
```

Reply:
Updated the code, util.format support multiple types so passing them as they are now, FYI this code is only applicable to Winston 2 https://github.com/winstonjs/winston/tree/2.x, string interpolation in newer version is handled before write is called.